cpp/src/umap/simpl_set_embed/algo.cuh

```diff
@@ -252,14 +252,12 @@ void optimize_layout(T* head_embedding,
 
   T rounding = create_gradient_rounding_factor<T>(head, nnz, head_n, alpha, stream_view);
 
-  MLCommon::FastIntDiv tail_n_fast(tail_n);
```

Comment:
This and similar changes might lead to a performance regression.

Reply:
Couldn't agree more. This is there for a reason. If we need to, we should be allowing a 64-bit int div instead of removing this altogether.

Reply:
I restored back the use of the `FastIntDiv` for the optimization kernels